src/utils/user.test.ts

```diff
@@ -0,0 +1,123 @@
+import { CutRoles } from '@/services/roles'
+import { getMockedAuthUser } from '@/tests/utils/models/user'
+import { expect } from '@jest/globals'
+import { Environment, Role, UserStatus } from '@prisma/client'
+import * as organizationModule from './organization'
+import { canBeUntrainedRole, findUserInfo, getEnvironmentRoles, getRoleToSetForUntrained, isAdmin } from './user'
+
+jest.mock('./organization', () => ({ canEditMemberRole: jest.fn() }))
+
+const mockCanEditMemberRole = organizationModule.canEditMemberRole as jest.Mock
+
+describe('userUtils functions', () => {
+  describe('isAdmin', () => {
+    test('should return true for ADMIN roles', () => {
+      expect(isAdmin(Role.ADMIN)).toBe(true)
+      expect(isAdmin(Role.SUPER_ADMIN)).toBe(true)
+    })
+
+    test('should return false for non-ADMIN role', () => {
+      expect(isAdmin(Role.GESTIONNAIRE)).toBe(false)
+      expect(isAdmin(Role.COLLABORATOR)).toBe(false)
+      expect(isAdmin(Role.DEFAULT)).toBe(false)
+      expect(isAdmin('OTHER_ROLE' as Role)).toBe(false)
+    })
+  })
+
+  describe('findUserInfo', () => {
+    test('should return correct arguments for user find info when user can edit member role', () => {
+      const user = getMockedAuthUser({ role: Role.ADMIN })
+      mockCanEditMemberRole.mockReturnValue(true)
+
+      const result = findUserInfo(user)
+      expect(result).toEqual({
+        select: {
+          user: {
+            select: {
+              email: true,
+              firstName: true,
+              lastName: true,
+              level: true,
+              status: true,
+              updatedAt: true,
+            },
+          },
+          role: true,
+          updatedAt: true,
+        },
+        where: { organizationVersionId: user.organizationVersionId },
+      })
+      expect(mockCanEditMemberRole).toHaveBeenCalledWith(user)
+    })
+
+    test('should return correct arguments for user find info when user cannot edit member role', () => {
+      const user = getMockedAuthUser({ role: Role.DEFAULT })
+      mockCanEditMemberRole.mockReturnValue(false)
+
+      const result = findUserInfo(user)
+      expect(result).toEqual({
+        select: {
+          user: {
+            select: {
+              email: true,
+              firstName: true,
+              lastName: true,
+              level: true,
+              status: true,
+              updatedAt: true,
+            },
+          },
+          role: true,
+          updatedAt: true,
+        },
+        where: { user: { status: UserStatus.ACTIVE }, organizationVersionId: user.organizationVersionId },
+      })
+      expect(mockCanEditMemberRole).toHaveBeenCalledWith(user)
+    })
+  })
+
+  describe('getEnvironmentRoles', () => {
+    test('should return CutRoles when NEXT_PUBLIC_DEFAULT_ENVIRONMENT is CUT', () => {
```

Comment:
Pas hyper important mais juste du coup vu qu'on utilise plus l'env (pareil juste en dessous)

```suggestion
    test('should return CutRoles when environment is CUT', () => {
```